packages/dataviews/src/view-actions.tsx

```diff
@@ -106,7 +110,9 @@ function ViewTypeMenu( {
 									return onChangeView( {
 										...view,
 										type: e.target.value,
-										layout: {},
+										fields:
+											defaultFields?.[ e.target.value ] ??
```

Comment:
This ties `view.fields` to `view.type`. However, when considering bundled or custom views, I see that it could be useful to have a few different `table` views that use different fields (same `view.type` but different `view.fields`), for example.

This is why I see this as a consumer decision, not dataview's. I'd remove the `defaultFields` from DataViews API and keep the `onChange` callback we have in consumers to augment the information based on the `type` change.

Reply:
I can see both ways but the current approach allows a clear cut (resets fields on each change) when you switch layouts which is a better default and if someone wants to keep or reuse fields, it's possible.

A small advantage too is that we don't have to repeat the same code in all consumers 

I'm open to change but I have a small preference as is. Would you be ok with trying this for some time ?

Reply:
Sure, not a blocker for me. I see how it can be a nice devexp touch.

The way I understand this prop is as a mechanism to augment or provide the "reset state" for the fields prop — any other view prop has a "natural" reset state (empty). Perhaps the main concern I have is that the name (`defaultFields`) doesn't really convey that concept.

Throwing some names in case some stick as self-explanatory:

- `onViewTypeReset`: this is the clearest I could come up with. It should also allow for providing a function besides the fields object. There's the question of whether other view props should be reset-configurable as well. I don't see a strong need for this, though perhaps `view.perPage` or future `view.density` could benefit from this as well. Ok to start with `fields` only.
- Other options: `viewTypeResetFieldConfig`, `viewTypeResetFields`, `fieldsToResetOnViewTypeChange`, `fieldsPerViewType`.

Reply:
I agree that there's potential for adding similar reset behaviors to other properties. At this point, the prop becomes `defaultViews` or something like that.

Reply:
Taking into account the conversations in the other threads, I think we should start this exploration with a better name than `defaultFields`: `fieldsRenderConfig` would work for me.

Reply:
but it's actually the default "fields" property of view. `view.fields`